Scripts/AssemblyInjections/Unity.Entities/ComponentTypeDependencyExtension.cs

```diff
@@ -0,0 +1,171 @@
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+/// <summary>
+/// A collection of extension methods to help calculate the scheduling dependencies on <see cref="ComponentType"/>s.
+/// </summary>
+[BurstCompatible]
+public static class ComponentTypeDependencyExtension
+{
+    private static UnsafeList<int> s_WriteTypeList_ScratchPad = new UnsafeList<int>(0, Allocator.Persistent);
```

Comment:
Should we allow it to allocate an initial amount like 16? Tiny amount of memory but avoids the realloc when this is first used and probably won't go over?

At 0, we'll realloc to 1, then 2, then 4 and then 8 each time we hit 1, 2, 3, and 5 items respectively

Reply:
I was following Unity's pattern for Systems. They do the same where it's 0 alloc. I think the idea is that if you never use this feature it doesn't take up any memory.

Reply:
Interesting. I wouldn't have thought that was a good idea but fair enough.

Reply:
Yea, realistically, most of the time on the first frame you'll get your max alloc settled and be done. It's only if you kick off some obscure job later that requires a bunch of dependencies that you'll see a new alloc.

If it was an issue we could always warm the extension ourselves by making a call with a large number of components.